File: Modul 6/Bahasa C/PRAK605_2310817120001_DinaIzzatiElfadheya.c
#include <stdio.h>
void matriks_persegi(int n, int matriks_A[n][n], int matriks_B[n][n], int hasil[n][n]) {
    for (int x = 0; x < n; x++) {
        for (int y = 0; y < n; y++) {
            hasil[x][y] = 0;
            for (int z = 0; z < n; z++) {
                hasil[x][y] += matriks_A[x][z] * matriks_B[z][y];
            }
        }
    }
}
void printmatriks(int n, int matriks[n][n]) {
    for (int x = 0; x < n; x++) {
        for (int y = 0; y < n; y++) {
            printf("%d ", matriks[x][y]);
        }
        printf("\n");
    }
}
int main() {
    int n;

    printf("Input\n");
    scanf("%d", &n);

    int matriks_A[n][n], matriks_B[n][n], hasil[n][n];

    printf("matriks A\n");
    for (int x = 0; x < n; x++) {
        for (int y = 0; y < n; y++) {
            scanf("%d", &matriks_A[x][y]);
        }
    }
    printf("matriks B\n");
    for (int x = 0; x < n; x++) {
        for (int y = 0; y< n; y++) {
            scanf("%d", &matriks_B[x][y]);
        }
    }
    matriks_persegi(n, matriks_A, matriks_B, hasil);

    printf("\nOutput");
    printf("\nMatriks AXB\n");
    printmatriks(n, hasil);
    return 0;
}

File: Modul 6/Bahasa Python/PRAK605_2310817120001_DinaIzzatiElfadheya.py
def matriks_persegi(n, matriks_A, matriks_B):
    hasil = [[0 for _ in range(n)] for _ in range(n)]
    
    for x in range(n):
        for y in range(n):
            hasil[x][y] = sum(matriks_A[x][k] * matriks_B[k][y] for k in range(n))
    return hasil

def print_matriks(matriks):
    for row in matriks:
        print(" ".join(map(str, row)))

def main():
    print("Input")
    n = int(input())
    
    print("Matriks A")
    matriks_A = [list(map(int, input().split())) 
        for _ in range(n)]

    print("Matriks B")
    matriks_B = [list(map(int, input().split())) 
        for _ in range(n)]

    hasil = matriks_persegi(n, matriks_A, matriks_B)

    print("\nMatriks AXB")
    print_matriks(hasil)

if __name__ == "__main__":
    main()
